config/puma.rb

```diff
@@ -25,7 +25,7 @@
   # worker per CPU, 1 thread per worker and tune it from there.
   #
   # https://edgeguides.rubyonrails.org/tuning_performance_for_deployment.html#puma
-  workers Concurrent.physical_processor_count
+  workers ENV.fetch("WEB_CONCURRENCY", Concurrent.physical_processor_count).to_i
```

Comment:
To match upstream Rails:
```suggestion
  workers Integer(ENV.fetch("WEB_CONCURRENCY") { Concurrent.physical_processor_count })
```

Reply:
If you're up for it!